torch/_inductor/kernel/flex_attention.py

```diff
@@ -630,10 +630,20 @@ def forward_block_mn(
 
 
 def _use_flex_decoding(query, kernel_options):
-    # Decide which kernel to use, return true if use flex decoding kernel.
-    return (
-        not kernel_options.get("FORCE_USE_FLEX_ATTENTION", False)
-    ) and V.graph.sizevars.evaluate_expr(sympy.Lt(query.get_size()[-2], 128))
+    """Decide which kernel to use, return true if use flex decoding kernel.
+    Note:
+       Since the number of splits is calculated based of the the number of batch and head dims
+       we need to ensure that the batch and head dims are statically known. Otherwise we just
+       use the main flex_attention kernel.
+    """
+    force_flex = kernel_options.get("FORCE_USE_FLEX_DECODING", False)
+    short_query_length = V.graph.sizevars.evaluate_expr(
+        sympy.Lt(query.get_size()[-2], 128)
+    )
+    V.graph.sizevars.statically_known_equals
+    static_batch = isinstance(query.get_size()[0], int)
+    static_num_heads = isinstance(query.get_size()[1], int)
+    return not force_flex and short_query_length and static_batch and static_num_heads
```

Comment:
This is probably wrong:
* If `FORCE_USE_FLEX_DECODING=True`, should we always use flex decoding kernel? I think we should respect it, even it errors out under some condition.
* If `FORCE_USE_FLEX_DECODING=False`, then we check the other conditions.
* You update the kernel option name from `FORCE_USE_FLEX_ATTENTION` to `FORCE_USE_FLEX_DECODING`, so `not force_flex` need to update to `force_flex`.

I think it should be:
```
return force_flex or short_query_length and static_batch and static_num_heads
```

Reply:
Yeah that name switch was an accident, just fixed